### PR TITLE
Stangec/layer arn check

### PR DESCRIPTION
**Pull request by cstanger**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -82,59 +67,89 @@ class AddLambdaInsights {
     if (typeof value === 'number') {
       return value;
     } else {
-      throw new Error(`LambdaInsightsVersion must be a number.`);
+      throw new Error(`LambdaInsights layer version must be a number.`);
```
Comment (iriskraja77) [inline index.js:70]: use ' instead

File: index.js [hunk 2]
```diff
@@ -82,59 +67,89 @@ class AddLambdaInsights {
...
+   */
+  async generateLayerARN(version) {
+    if (version) {
+      try {
+        const layerVersionInfo = await this.provider.request('Lambda', 'getLayerVersionByArn', {
+          Arn: `arn:aws:lambda:${this.region}:580247275435:layer:LambdaInsightsExtension:${version}`,
+        });
+        return layerVersionInfo.LayerVersionArn;
+      } catch (err) {
+        throw new Error(
```
Comment (iriskraja77) [inline index.js:87]: If the error is something else, this error would be thrown and the message can be misleading. Could you check if err.message = 'not exists' then throw this error.

File: index.test.js [hunk 3]
```diff
@@ -22,26 +35,79 @@ test('addLambdaInsights throws for unknown region', () => {
   const task = () => plugin.addLambdaInsights();
 
   // assert
-  expect(task).toThrow(Error);
```
Comment (iriskraja77) [inline index.test.js:25]: You can also do this: `await expect(task)).rejects.toThrow(message)`

File: index.js (hunk 2 above)
Comment (cstanger) [reply index.js:87]: I would not 'hide' any other errors thrown by the other code beside the function generateLayerARN, as the explicit error message would be the most direct one. What do you think?

Comment (iriskraja77) [reply index.js:87]: Yeah I think it should be fine actually.

File: index.js [hunk 4]
```diff
@@ -82,59 +67,89 @@ class AddLambdaInsights {
...
+      throw new Error('lambdaInsightsVersion version must be a number.');
     }
   }
 
+  /**
+   * Generates a valid Lambda Insights Layer ARN for your Region
+   * @param  {number} version Value to check
+   * @return {string} Lambda Insights Layer ARN
+   */
+  async generateLayerARN(version) {
```
Comment (jaredcnance) [inline index.js:79]: Should we also allow overriding the account id? I'm thinking of a case when a new version is released in an opt-in region but it hasn't been updated here.

File: index.js (hunk 2 above)
Comment (jaredcnance) [reply index.js:87]: What if they get throttled or service unavailable or any error that may actually succeed if retried? Aren't we hiding the underlying error? Should we at least log the SDK error?

**Review by jaredcnance — APPROVED** [review]

File: index.js (hunk 4 above)
Comment (cstanger) [reply index.js:79]: Very nice idea, let's make a complete new Feature Request out of it.

File: index.js (hunk 2 above)
Comment (cstanger) [reply index.js:87]: Somethink like this?

```suggestion
          if (err.code==='AccessDeniedException') {
          throw new Error(
              `LambdaInsights layer version '${version}' ` +
            `does not exist within your region '${this.region}'.`);
        } else {
          throw err;
        }
```

Comment (jaredcnance) [reply index.js:87]: LGTM

**Pull request merged**